src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -411,6 +425,64 @@ def visit_span(self, span: Span):
             )
 
 
+class RenderBlockingAssetSpanDetector(PerformanceDetector):
+    __slots__ = ("stored_issues", "fcp", "transaction_start")
+
+    settings_key = DetectorType.RENDER_BLOCKING_ASSET_SPAN
+
+    def init(self):
+        self.stored_issues = {}
+        self.transaction_start = timedelta(seconds=self.event().get("transaction_start", 0))
+        self.fcp = None
+
+        # Only concern ourselves with transactions where the FCP is within the
+        # range we care about.
+        fcp_hash = self.event().get("measurements", {}).get("fcp", {})
+        if "value" in fcp_hash and ("unit" not in fcp_hash or fcp_hash["unit"] == "millisecond"):
+            fcp = timedelta(milliseconds=fcp_hash.get("value"))
+            fcp_minimum_threshold = timedelta(
+                milliseconds=self.settings.get("fcp_minimum_threshold")
+            )
+            fcp_maximum_threshold = timedelta(
+                milliseconds=self.settings.get("fcp_maximum_threshold")
+            )
+            if fcp >= fcp_minimum_threshold and fcp < fcp_maximum_threshold:
+                self.fcp = fcp
+
+    def visit_span(self, span: Span):
+        if not self.fcp:
+            return
+
+        op = span.get("op", None)
+        allowed_span_ops = self.settings.get("allowed_span_ops")
+        if op not in allowed_span_ops:
+            return False
+
+        if self._is_blocking_render(span):
+            span_id = span.get("span_id", None)
+            fingerprint = fingerprint_span(span)
+            if span_id and fingerprint:
+                self.stored_issues[fingerprint] = PerformanceSpanIssue(span_id, op, [span_id])
+
+        # If we visit a span that starts after FCP, then we know we've already
+        # seen all possible render-blocking resource spans.
+        span_start_timestamp = timedelta(seconds=span.get("start_timestamp", 0))
+        fcp_timestamp = self.transaction_start + self.fcp
+        if span_start_timestamp >= fcp_timestamp:
+            # Early return for all future span visits.
+            self.fcp = None
```

Comment:
Yeah we can either have `visit_span` return, or we set it up so `visit_span` is indirectly called from a base function which checks some base property like `PerformanceDetector.isDisabled` so each class has more control.